src/components/namestable/worker-edit/worker-edit.component.tsx

```diff
@@ -0,0 +1,122 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Grid } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import classNames from "classnames/bind";
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { ContractType, WorkerType } from "../../../common-models/worker-info.model";
+import { WorkerActionCreator } from "../../../state/reducers/worker.action-creator";
+import { Button } from "../../common-components";
+import { useWorkerInfo, WorkerInfo } from "../use-worker-info";
+import { WorkerContractTypeDependentWorkNormSelectror } from "./worker-contract-type-dependent-worknorm-selector.component";
+import { WorkerContractTypeSelector } from "./worker-contract-type-selector.component";
+import { WorkerEditComponentOptions, WorkerEditComponentMode } from "./worker-edit.models";
+import { WorkerNameEditField } from "./worker-name-edit-field.components";
+import { WorkerWorkerTypeSelector } from "./worker-position-selector.component";
+
+const useStyles = makeStyles({
+  container: {
+    height: "93%",
+  },
+});
+
+export function WorkerEditComponent({
+  name,
+  mode,
+  setOpen,
+}: WorkerEditComponentOptions): JSX.Element {
+  const classes = useStyles();
+
+  const dispatcher = useDispatch();
+  const { workerInfo, setWorkerInfo } = useWorkerInfo(name);
+  const [isWorkerNameValid, setIsWorkerNameValid] = useState(true);
+  const [isWorkerTimeValid, setIsWorkerTimeValid] = useState(true);
+  const [isContractTypeValid, setIsContractTypeValid] = useState(true);
+  const [isWorkerTypeValid, setIsWorkerTypeValid] = useState(true);
+
+  useEffect(() => {
+    if (mode === WorkerEditComponentMode.EDIT) {
+      workerInfo.previousWorkerName = name;
+    }
+  }, [mode, workerInfo, name]);
+  //#region event handlers
```

Comment:
komentarze regionów w tym pliku są potrzebne?

Reply:
Nie są
Nie mniej jednak preferuję żeby one były ponieważ moim zdaniem zwiększają czytelność kodu dzieląc go na logiczne sekcje. 

Ponadto to ma taką zaletę, że wszystko zawarte w regionie może być zwinięte
![image](https://user-images.githubusercontent.com/21079319/111532449-f6faa480-8765-11eb-9601-20aa583cc704.png)


PS
W intellij to chyba wymaga dodatkowej konfiguracji